api/src/test/java/jakarta/data/metamodel/expression/StringLiteralTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.expression;
+
+import jakarta.data.metamodel.constraint.Like;
+import jakarta.data.metamodel.constraint.NotLike;
+import jakarta.data.metamodel.restrict.BasicRestriction;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class StringLiteralTest {
+
+    interface _SimpleEntity {
+        StringLiteral<SimpleEntity> name = StringLiteral.of("name");
+    }
+
```

Comment:
If this was supposed to be a static metamodel class (the naming pattern looks like it), it is incorrect usage to have something other than an Attribute or String constant in it,
```suggestion
```